src/test/java/redis/clients/jedis/tests/commands/AllKindOfValuesCommandsTest.java

```diff
@@ -868,4 +871,37 @@ public void sendCommandTest(){
     assertEquals("PONG", SafeEncoder.encode((byte[]) jedis.sendCommand(PING)));
   }
 
+
+  @Test
+  public void encodeCompleteResponse(){
+    HashMap<String,String> entry = new HashMap<>();
+    entry.put("foo", "bar");
+    jedis.xadd( "mystream", StreamEntryID.NEW_ENTRY, entry );
+    String status = jedis.xgroupCreate("mystream", "mygroup", null, false);
+
+    Object obj = jedis.sendCommand(XINFO, "STREAM", "mystream");
+    ArrayList encodeObj =  (ArrayList)SafeEncoder.encodeObject(obj);
```

Comment:
This and subsequent type casts should be `List` instead of `ArrayList`.

Reply:
@sazzad16 I have update the test